diff --git a/cohesion.install b/cohesion.install
index e93ecf9..2b75486 100755
--- a/cohesion.install
+++ b/cohesion.install
@@ -19,7 +19,7 @@ use Drupal\Core\Installer\InstallerKernel;
  */
 function cohesion_requirements($phase) {
   $requirements = [];
-  if (\Drupal::moduleHandler()->moduleExists('acquia_lift') && !\Drupal::moduleHandler()->moduleExists('sitestudio_contenthub_publisher')) {
+  if (\Drupal::moduleHandler()->moduleExists('acquia_contenthub') && !\Drupal::moduleHandler()->moduleExists('sitestudio_contenthub_publisher')) {
     $settings = ['@link' => 'https://github.com/acquia/sitestudio_contenthub'];
     $requirements['cohesion'] = array(
       'title' => t('Site Studio'),
